fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -461,14 +474,28 @@ private void maybePerformScan() throws IOException {
                     } else if (exhausted) {
                         nextResult = RecordCursorResult.exhausted();
                     } else if (limitRemaining <= 0) {
-                        RecordCursorContinuation continuationFromDoc = LuceneCursorContinuation.fromScoreDoc(searchAfter, partitionId, partitionTimestamp);
+                        final Tuple primaryKey;
+                        if (nextResult != null && nextResult.get() != null) {
+                            primaryKey = nextResult.get().getPrimaryKey();
+                        } else {
+                            primaryKey = null;
+                        }
+                        RecordCursorContinuation continuationFromDoc = LuceneCursorContinuation.fromScoreDoc(
+                                searchAfter,
+                                partitionId,
+                                partitionKey,
+                                primaryKey);
```

Comment:
This seems incorrect, we should be getting the primaryKey from the `searchAfter`.

Reply:
We won't need to pass it explicitly (will grab it from the `ScoreDoc`)